ActiveLabeler-main/ActiveLabeler.py

```diff
@@ -0,0 +1,181 @@
+                mean = 0.5
+                sigma = 0.2
+                return [gaussian(conf_i, mean, sigma) for conf_i in confidences]
+
+            def get_samples(image_list, confidences, number_of_samples):
+                sample_wts = get_gaussian_weights(confidences)
+                sample_wts_norm = np.array(sample_wts)
+                sample_wts_norm /= sample_wts_norm.sum()
+                picked_samples = np.random.choice(
+                    image_list, replace=False, size=number_of_samples, p=sample_wts_norm
+                )
+                return picked_samples
+
+            def get_index(samples, dataset):
+                data_index = []
+                for i in samples:
+                    data_index.append(dataset.index(i))
+                data_index = np.array(data_index)
+                return dataset[data_index]
+
+            selected_samples = get_samples(self.images_path, query, N)
+            # selected_embeddinds = get_index(selected_samples, self.embeddings)
+            return selected_samples
+        elif name == "uncertainty":
+            tmp_query = np.array([query[i][0] for i in range(len(query))])
+            difference_array = np.absolute(tmp_query - 0.5)
+            return difference_array.argsort()[:N]
+        elif name == "uncertainty_balanced":
+            tmp_query = np.array([query[i][0] for i in range(len(query))])
+            difference_array = tmp_query - 0.5
+            sorted_diff = difference_array.argsort()
+            idx_0 = np.absolute(difference_array).argsort()[0]
+            idx_sorted_0 = list(sorted_diff).index(idx_0)
+
+            #todo logs
+            print("tmp_query- probs - sorted", sorted(tmp_query))
+            print("diff arr ", difference_array)
+            print("diff arr sorted ", np.sort(difference_array))
+            print("middle value idx", idx_sorted_0)
+            print("middle value ", difference_array[idx_sorted_0])
+
+            N = N - 1
+            # take n/2 from either side of index_sorted_0, less ele
+            print("splitting idices", (idx_sorted_0 - (N // 2) - 1), idx_sorted_0)
+            print("splitting idices", idx_sorted_0 + 1, (idx_sorted_0 + (N // 2)))
+            tmp1 = sorted_diff[(idx_sorted_0 - (N // 2)) : idx_sorted_0]
+            tmp2 = sorted_diff[idx_sorted_0 + 1 : (idx_sorted_0 + (N // 2)) + 1]
+            print("tmp1", tmp1)
+            print("tmp2", tmp2)
+            print(
+                f"uncertainity balanced: {len(tmp1)} + {len(tmp2)} = {len(tmp1) + len(tmp2)}"
+            )
+            # returning the corresponding indexes tmp1 + tmp2
+            tmp3 = []
+            for i in tmp1:
+                tmp3.append(i)
+            for i in tmp2:
+                tmp3.append(i)
+            tmp3.append((idx_0))
+            return tmp3
+        elif name == "random":
+            return [random.randrange(0, len(query), 1) for i in range(N)]
+        elif name == "positive":
```

Comment:
It looks like name can only have a defined set of values. Consider creating an [enum](https://docs.python.org/3/library/enum.html) here and also call it `strategy` instead of `name`, as `name` conveys that it's a more freeform field.